### DISCUSSION TITLE
User-Customized game shortcuts

### DISCUSSION THREAD

**Discussion by brine** [octgn/OCTGN#2119] [Feature Planning]
from #953.  A game options dialog window (probably linked from the game manager) fits this role well.

The biggest obstacle I see with this is that our game definitions don't enforce any sort of unique value within the card/group actions.  So if we want to actually save the changes, we need some way to reference that action definition in the config/settings file. 

Alternatively, the dialog window could just directly modify the hotkey in the game definition.  This is a bad idea though since a game update would nuke any user-modified shortcuts.

Any ideas for tackling this issue would be great.

**Comment by BenMatteson** [comment]
I mean, if we did it with a dialog we control we could include whatever validation we need. I feel like the real issue would be detecting and overriding the game's hotkeys, but maybe it's easier than I expect, I've not really messed with any of that code.

**Comment by brine** [comment]
we could hash the action's name and use that as the key for the config file, but that assumes that the names are all unique already.

Also, if the game developer changes the name of the action, it'll break that hotkey binding.  Which I guess is unavoidable regardless.
